content/client-lib-development-guide/features.textile

```diff
@@ -859,6 +864,85 @@ h3(#channel-states-operations). @RealtimeChannel.state@ effects on channel opera
 
 </table>
 
+h2(#push-notifications). Push notifications
+
+* @(RSH1)@ @Push#publish(recipient, data)@ performs a HTTP request to "/push/publish":/rest-api/#push-publish
+* @(RSH2)@ The following should only apply to platforms that support receiving push notifications:
+** @(RSH2a)@ @Push#activate@ sends a @CalledActivate@ event to "the state machine":#RSH3.
+** @(RSH2b)@ @Push#deactivate@ sends a @CalledDeactivate@ event to "the state machine":#RSH3.
+** @(RSH2c)@ Whenever, from the platform's APIs, details for sending push notifications to the local device (e. g. GCM's registration token) is received, a @GotPushDeviceDetails@ event is sent to "the state machine":#RSH3.
+
+h3(#activation-state-machine). Activation state machine
+
+* @(RSH3)@ In platforms that support receiving push notifications, in order to connect the device's push features with Ably's, the library must perform the process described in the following abstract state machine. While this process should be implemented in whatever way better fits the concrete platform, it should be taken into account that its lifetime is that of the the _app_ that runs it, which outlives that of the @Rest@ instance or (typically) the process running the app. This typically forces some kind of on-disk storage to which the state machine's state must be persisted, so that it can be recovered later by new instances and processes running the app triggered by external events.
```

Comment:
The activation state machine should be a singleton to guarantee consistency of the persistent data. Now, there's a problem because, i.e. in iOS cannot guarantee the singleton if the user can create multiple rest clients with different keys. Which one will the state machine use? A device can only be registered to one Ably key? An app could use different Ably keys?